model/modeldecoder/generator/ninterface.go

```diff
@@ -87,19 +100,19 @@ if utf8.RuneCountInString(t) %s %s{
 	return fmt.Errorf("'%s': validation rule '%s(%s)' violated")
 }
 `[1:], ruleMinMaxOperator(maxRule.name), maxRule.value, jsonName(f), maxRule.name, maxRule.value)
+			} else if targetTypeRule != (validationRule{}) {
```

Comment:
```suggestion
			} else if targetTypeRule.value == "int" {
```

I realise it's the only possible value due to the previous check, but I think this might make the connection between the rule and the Atoi below a bit clearer.